src/middlewared/middlewared/plugins/disk_/zfs_guid.py

```diff
@@ -38,38 +33,32 @@ async def sync_zfs_guid(self, pool_id_or_pool):
             if vdev["type"] == "DISK" and vdev["disk"] is not None:
                 disk_to_guid[vdev["disk"]] = vdev["guid"]
 
+        events = []
         for disk in await self.middleware.call("disk.query", [], {"extra": {"include_expired": True}}):
             guid = disk_to_guid.get(disk["devname"])
             if guid is not None and guid != disk["zfs_guid"]:
-                logger.debug("Setting disk %r zfs_guid %r", disk["identifier"], guid)
+                self.logger.debug("Setting disk %r zfs_guid %r", disk["identifier"], guid)
+                events.append({"id": disk["identifier"], "zfs_guid": guid})
                 await self.middleware.call(
-                    "datastore.update", "storage.disk", disk["identifier"], {"zfs_guid": guid}, {"prefix": "disk_"},
+                    "datastore.update", "storage.disk", disk["identifier"],
+                    {"zfs_guid": guid}, {"prefix": "disk_", "send_event": False},
                 )
             elif disk["zfs_guid"]:
                 devname = disk_to_guid.inv.get(disk["zfs_guid"])
                 if devname is not None and devname != disk["devname"]:
-                    logger.debug("Removing disk %r zfs_guid as %r has it", disk["identifier"], devname)
+                    self.logger.debug("Removing disk %r zfs_guid as %r has it", disk["identifier"], devname)
+                    events.append({"id": disk["identifier"], "zfs_guid": None})
                     await self.middleware.call(
-                        "datastore.update", "storage.disk", disk["identifier"], {"zfs_guid": None}, {"prefix": "disk_"},
+                        "datastore.update", "storage.disk", disk["identifier"],
+                        {"zfs_guid": None}, {"prefix": "disk_", "send_event": False},
                     )
 
+        if events:
+            self.middleware.send_event('disk.query', 'CHANGED', fields=events)
```

Comment:
This does not match the signature of events that were sent by the original code (it was an event per single row, with `id` field set). Same for `disk.sync_all`.

Reply:
I understand but I don't know of a better way of doing this. I figured the webUI is going to look for the difference. Maybe the `id` key can be `BATCH` or something and they can catch that.

Reply:
@yocalebo can you send multiple events in a `for` loop?

Reply:
oh...you're right. I thought `middleware.send_event` was doing the `disk.query` and was trying to send in 1 call but that's not the case. I'll fix.